### ISSUE TITLE
Automatic scope definition can break applications because of naming issues

### ISSUE THREAD

**Issue by klyonrad** [betterup/boolean_timestamps#7]
We get an error upon upgrading to boolean_timestamps 1.1.0

Sorry @jordanbyron but I think you introduced a bug with [your latest feature addition](https://github.com/betterup/boolean_timestamps/pull/4) 😿

# Reproduction

## model code

```ruby

class SomeModel < ApplicationRecord
  boolean_timestamps :loaded_at
end
```

## error

leads to 

```
ArgumentError: You tried to define a scope named "loaded" on the model "SomeModel", but ActiveRecord::Relation already defined an instance method with the same name.
```

So boolean_timestamps tries to overwrite this method: https://www.rubydoc.info/docs/rails/ActiveRecord%2FRelation:loaded

----

Rails has some safeguards during scope definitions, [See here](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/scoping/named.rb#L173)

I propose to either respect those safeguards or make the scope definitions Opt-In/Opt-Out

**Comment by jordanbyron** [comment]
Ah yes that is clearly a problem. Thank you so much for letting me know. I like the idea of making these scope definitions opt-out for instances where they clash with built in methods. Is that a feature you'd be comfortable working on?

In the meantime it should be safe to downgrade to `1.0.x` until this is resolved.

**Comment by klyonrad** [comment]
> Ah yes that is clearly a problem. Thank you so much for letting me know. I like the idea of making these scope definitions opt-out for instances where they clash with built in methods. Is that a feature you'd be comfortable working on?

No, sorry, I could not spare the working time resources on this. Personally

I think it would be faster to add an option parameter [kind of like flag shih tzu does it](https://github.com/pboling/flag_shih_tzu#generated-named-scopes). With such an option available gem users would at least be able to help themselves.

**Comment by jordanbyron** [comment]
> I think it would be faster to add an option parameter [kind of like flag shih tzu does it](https://github.com/pboling/flag_shih_tzu#generated-named-scopes). With such an option available gem users would at least be able to help themselves.

That's exactly what I had in mind! I totally understand that you don't have time to implement this feature. Thanks for the report! 😄
